src/app/CommandHandler.cpp

```diff
@@ -801,6 +841,21 @@ CommandHandler::Handle::Handle(CommandHandler * handle)
     }
 }
 
+CHIP_ERROR CommandHandler::FinalizeInvokeResponseMessageAndReadyNext()
+{
+    ReturnErrorOnFailure(FinalizeInvokeResponseMessage(/* aHasMoreChunks = */ true));
+    CHIP_ERROR err = AllocateBuffer();
+    if (err == CHIP_NO_ERROR)
+    {
+        VerifyOrDie(mState == State::NewResponseMessage);
```

Comment:
why is this being checked only if we do not have an allocation error? is it ok to be in a different state as long as allocation fails? can AllocateBuffer change state?